libraries/Wire/src/Wire.h

```diff
@@ -54,6 +54,7 @@ class TwoWire : public Stream
     void begin(int);
     void end();
     void setClock(uint32_t);
+    void setTimeoutInMicros(uint32_t);
```

Comment:
I think the Stream timeout is intended for *reading* and could still be used with Wire if you do e.g. `readStringUntil()`. It does not really make much sense, since the Wire library will always only have a fixed packet available.

I would *not* reuse this method for the TWI timeout, since that would only increase the confusion. Also, I suspect it is not virtual, so then calling `Wire.setTimeout()` would be different from calling `((Stream&)Wire).setTimeout()` which might be more confusing.

Instead, I would suggest renaming the new `setTimeoutInMicros()` to `setWireTimeout()` to make it more explicit. I'm also not so sure if `InMicros` is really a needed addition, I do not think other methods use the same convention (though it is of course nice and explicit). `setWireTimeoutUs()` maybe?

Reply:
>I think the Stream timeout is intended for reading ...

Indeed, it is used for reading. Currently **Wire** library doesn't use those read methods (from Stream class) with timeout.

>Also, I suspect it is not virtual...

Indeed, it is not virtual

>Instead, I would suggest renaming the new ````setTimeoutInMicros()```` to ````setWireTimeout()```` to make it more explicit.

````setWireTimeout```` seems to more self-explanatory.
````Stream::setTimeout```` requires milliseconds however we plan for ````Wire::setWireTimeout```` to accept microseconds. Would that be a bit confusing? So maybe ````setWireTimeoutUs```` or ````setWireTimeoutMicros````? I have no idea.

Reply:
c45ac98 renames the set timeout function to `setWireTimeoutUs` to avoid confustion with the Stream timeout